src/humanpy/initialize.py

```diff
@@ -1,120 +1,141 @@
+from prpy.base.endeffector import EndEffector
+from prpy.base.manipulator import Manipulator
+from prpy.base.robot import Robot
+from prpy.bind import bind_subclass
+from prpy.planning.base import Sequence
+from prpy.planning.cbirrt import CBiRRTPlanner
+from prpy.planning.vectorfield import VectorFieldPlanner
+from prpy.planning.snap import SnapPlanner 
+from prpy.planning.retimer import HauserParabolicSmoother
+import numpy
+
+logger = logging.getLogger('humanpy')
+
+def initialize(attach_viewer=False, sim=True, user_id='human', env=None):
+    """Initialize the Human Robot"""
+    
+    prpy.logger.initialize_logging()
+    
+    from catkin.find_in_workspaces import find_in_workspaces
+    package_name = 'humanpy'
+    directory = 'ordata/dae'
+    objects_path = find_in_workspaces(
+        search_dirs=['share'],
+        project=package_name,
+        path=directory,
+        first_match_only=True)
+    if len(objects_path) == 0:
+        print('Can\'t find directory %s/%s' % (package_name, directory))
+        sys.exit()
+    else:
+        print objects_path # for me this is '/home/spelle/storage/ros_test_ws/src/humanpy/ordata/dae/man1.dae'
+        objects_path = objects_path[0]
+    
+    if not env:
+        env = Environment()
+
+    #Setup Manipulators
+    with env:
+        robot_file = os.path.join(objects_path, 'man1.dae')
+        robot = env.ReadKinBodyXMLFile(robot_file)
+        robot.SetName(user_id)              #needed in order to have different humans in the same env
+        env.AddKinBody(robot) 
+
+        robot.left_arm = robot.GetManipulator('leftarm')
+        robot.left_arm.hand = robot.left_arm.GetEndEffector()
+        robot.left_hand = robot.left_arm.hand
+
+        robot.right_arm = robot.GetManipulator('rightarm')
+        robot.right_arm.hand = robot.right_arm.GetEndEffector()
+        robot.right_hand = robot.right_arm.hand
+
+        bind_subclass(robot, Robot, robot_name=user_id)
+        bind_subclass(robot.left_arm, Manipulator)
+        bind_subclass(robot.right_arm, Manipulator)
+    
+        bind_subclass(robot.left_arm.hand, HumanHand, manipulator=robot.left_arm, sim=True)
+        bind_subclass(robot.right_arm.hand, HumanHand, manipulator=robot.right_arm, sim=True)
+        
+        #Setup Controller
+        with env:
+            controller_dof_indices = []
+            controller_dof_indices.extend(robot.left_arm.GetArmIndices())
+            controller_dof_indices.extend(robot.right_arm.GetArmIndices())
+
+            robot.right_arm.controller = robot.AttachController(
+                    name=robot.right_arm.GetName(), args='',
+                    dof_indices=robot.right_arm.GetArmIndices(), affine_dofs=0, 
+                    simulated=sim)
+            robot.left_arm.controller = robot.AttachController(
+                    name=robot.left_arm.GetName(), args='',
+                    dof_indices=robot.left_arm.GetArmIndices(), affine_dofs=0, 
+                    simulated=sim)

```

Comment:
Yes, the source code of that function is a good reference. You can use `RaveCreateController` to create an `IdealController` and add it to the multi-controller using `robot.multicontroller.AttachController`.

However, I am beginning to wonder if this is necessary. Do you ever execute trajectories on the humans in `humanpy`? Or call `SetDesired` on their controller? If not, there's no reason to register a controller at all.